### PR TITLE
[openwebnet] handle Thermo Central Unit monitoring messages

### PR DESCRIPTION
**Pull request by aconte80**
Enhance OpenWebNet binding to support thermoregulation with 4/99 zone central unit, e.g. BTicino 3550 model.

The Central Unit provides some information on the on-board display:

- at least one probe in OFF mode (WHAT = AT_LEAST_ONE_PROBE_OFF)
- at least one probe in PROTECTION mode (WHAT = AT_LEAST_ONE_PROBE_ANTIFREEZE)
- at least one probe in MANUAL mode (WHAT = AT_LEAST_ONE_PROBE_MANUAL)
- a generic FAILURE was detected on the system (WHAT = FAILURE_DISCOVERED)

For each case a specific read-only channel was added.

### CODE REVIEW

**Comment by lolodomo** [comment]
Conflicts to be resolved first.

File: bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetThermoregulationHandler.java [hunk 1]
```diff
@@ -438,7 +484,31 @@ private void updateCURemoteControlStatus(String status) {
 
     private void updateCUBatteryStatus(String status) {
         updateState(CHANNEL_CU_BATTERY_STATUS, new StringType(status));
-        logger.debug("updateCUBatteryStatus(): {}", status);
+
+        if (status == CU_BATTERY_KO) // do not log default value (which is automatically setted)
+            logger.debug("updateCUBatteryStatus(): {}", status);
```
Comment (lolodomo) [inline bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetThermoregulationHandler.java:489]: Please surround with {}

File: bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetThermoregulationHandler.java [hunk 2]
```diff
@@ -438,7 +484,31 @@ private void updateCURemoteControlStatus(String status) {
...
+    }
+
+    private void updateCUFailureDiscovered(OnOffType status) {
+        updateState(CHANNEL_CU_FAILURE_DISCOVERED, status);
+
+        if (status == OnOffType.ON) // do not log default value (which is automatically setted)
+            logger.debug("updateCUFailureDiscovered(): {}", status);
```
Comment (lolodomo) [inline bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetThermoregulationHandler.java:496]: Please surround with {}

File: bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetThermoregulationHandler.java [hunk 3]
```diff
@@ -355,6 +368,39 @@ private void updateModeAndFunction(Thermoregulation tmsg) {
...
+            } else if (mode == Thermoregulation.OperationMode.MANUAL) {
+                probesInProtection.remove(tmsg.getWhere().value());
+                probesInOFF.remove(tmsg.getWhere().value());
+                if (probesInManual.add(tmsg.getWhere().value())) {
+                    logger.debug("atLeastOneProbeInManual: added WHERE ---> {}", tmsg.getWhere());
+                }
+            }
+
+            if (probesInOFF.size() == 0) {
+                updateCUAtLeastOneProbeOFF(OnOffType.OFF);
```
Comment (lolodomo) [inline bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetThermoregulationHandler.java:394]: Rather use isEmpty()

File: bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetThermoregulationHandler.java [hunk 4]
```diff
@@ -355,6 +368,39 @@ private void updateModeAndFunction(Thermoregulation tmsg) {
...
+            }
+            if (probesInProtection.size() == 0) {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetThermoregulationHandler.java:396]: Rather use isEmpty()

File: bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetThermoregulationHandler.java [hunk 5]
```diff
@@ -355,6 +368,39 @@ private void updateModeAndFunction(Thermoregulation tmsg) {
...
+                updateCUAtLeastOneProbeProtection(OnOffType.OFF);
+            }
+            if (probesInManual.size() == 0) {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetThermoregulationHandler.java:399]: Rather use isEmpty()

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.openwebnet/README.md [hunk 6]
```diff
@@ -291,11 +294,15 @@ Number:Power        iCENTRAL_Tb           "Power [%.0f %unit%]" { channel="openw
...
-Number iCU_3550_spn       "Scenario Program"  (gCentralUnit) { channel="openwebnet:bus_thermo_cu:mybridge:CU_3550:scenarioProgram" } 
+String iCU_3550_remote    "Remote Control"                    (gCentralUnit) { channel="openwebnet:bus_thermo_cu:mybridge:CU_3550:remoteControl" }
+String iCU_3550_battery   "Battery Status"                    (gCentralUnit) { channel="openwebnet:bus_thermo_cu:mybridge:CU_3550:batteryStatus" }
+String iCU_3550_mode      "Mode"                              (gCentralUnit) { channel="openwebnet:bus_thermo_cu:mybridge:CU_3550:mode" }
+Number iCU_3550_wpn       "Weekly Program"                    (gCentralUnit) { channel="openwebnet:bus_thermo_cu:mybridge:CU_3550:weeklyProgram" } 
+Number iCU_3550_spn       "Scenario Program"                  (gCentralUnit) { channel="openwebnet:bus_thermo_cu:mybridge:CU_3550:scenarioProgram" } 
+String iCU_3550_func      "Function"                          (gCentralUnit) { channel="openwebnet:bus_thermo_cu:mybridge:CU_3550:function" }
+Switch iCU_3550_at1off    "At least one probe in OFF"         (gCentralUnit) { channel="openwebnet:bus_thermo_cu:mybridge:CU_3550:atLeastOneProbeOFF" }
```
Comment (lolodomo) [inline bundles/org.openhab.binding.openwebnet/README.md:303]: Please also add a line for new channel failureDiscovered.

**Review by lolodomo — CHANGES_REQUESTED** [review]

**Comment by lolodomo** [comment]
@mvalla : is it ok for you ?

**Review by lolodomo — APPROVED** [review]
LGTM

File: bundles/org.openhab.binding.openwebnet/src/main/resources/OH-INF/thing/channels.xml [hunk 7]
```diff
@@ -281,6 +281,34 @@
...
+		<item-type>Switch</item-type>
+		<label>At least one probe OFF</label>
+		<description>At least one probe OFF indicator in Central Unit (read only)</description>
+		<state readOnly="true"></state>
+	</channel-type>
+
+	<channel-type id="atLeastOneProbeProtection" advanced="true">
+		<item-type>Switch</item-type>
+		<label>At least one probe in Protection</label>
+		<description>At least one probe in Protection (Anti Freeze/Thermal Protection) indicator in Central Unit (read only)</description>
```
Comment (mvalla) [inline bundles/org.openhab.binding.openwebnet/src/main/resources/OH-INF/thing/channels.xml:301]:
```suggestion
		<description>At least one probe in PROTECTION (Anti Freeze/Thermal Protection) indicator in Central Unit (read only)</description>
```

File: bundles/org.openhab.binding.openwebnet/src/main/resources/OH-INF/thing/BusThermoCentralUnit.xml [hunk 8]
```diff
@@ -18,6 +18,10 @@
 			<channel id="remoteControl" typeId="remoteControl"/>
 			<channel id="batteryStatus" typeId="batteryStatus"/>
 			<channel id="function" typeId="functionCentralUnit"/>
+			<channel id="failureDiscovered" typeId="failureDiscovered"/>
+			<channel id="atLeastOneProbeOFF" typeId="atLeastOneProbeOFF"/>
```
Comment (mvalla) [inline bundles/org.openhab.binding.openwebnet/src/main/resources/OH-INF/thing/BusThermoCentralUnit.xml:22]: What about changing `atLeastOneProbeOFF ` -> `atLeastOneProbeOff ` similar to the other channels?

File: bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/OpenWebNetBindingConstants.java [hunk 9]
```diff
@@ -133,6 +133,10 @@
     public static final String CHANNEL_CU_BATTERY_STATUS = "batteryStatus";
     public static final String CHANNEL_CU_WEEKLY_PROGRAM_NUMBER = "weeklyProgram";
     public static final String CHANNEL_CU_SCENARIO_PROGRAM_NUMBER = "scenarioProgram";
+    public static final String CHANNEL_CU_FAILURE_DISCOVERED = "failureDiscovered";
+    public static final String CHANNEL_CU_AT_LEAST_ONE_PROBE_OFF = "atLeastOneProbeOFF";
```
Comment (mvalla) [inline bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/OpenWebNetBindingConstants.java:137]: What about changing `atLeastOneProbeOFF ` -> `atLeastOneProbeOff ` similar to the other channels?
see other comment below

File: bundles/org.openhab.binding.openwebnet/README.md [hunk 10]
```diff
@@ -188,7 +188,10 @@ The (optional) Central Unit can be configured defining a `bus_themo_cu` Thing.
...
 | `scenarioProgram`            | `bus_thermo_cu`                        | Number             | The program number (`1`, `2`, .. ,  `16`) when Central Unit mode is `SCENARIO`                                                                                                          | R/W        | N        |
-
+| `failureDiscovered`          | `bus_thermo_cu`                        | Switch             | Indicates if a Failure was discovered by the Central Unit: `ON`, `OFF`                                                                                                                  | R          | Y        |
+| `atLeastOneProbeOFF`         | `bus_thermo_cu`                        | Switch             | Indicates if at least one probe is in OFF mode: `ON`, `OFF`                                                                                                                             | R          | Y        |
```
Comment (mvalla) [inline bundles/org.openhab.binding.openwebnet/README.md:192]: What about changing `atLeastOneProbeOFF ` -> `atLeastOneProbeOff ` similar to the other channels?
see other comment below

**Review by mvalla — CHANGES_REQUESTED** [review]
I would only change the capital OFF in the name of the channel, just to make it similar to the other channels.  For the rest it looks good, excellent improvement thanks!

**Comment by aconte80** [comment]
@mvalla I've updated string upper/lowercase as suggested

**Comment by mvalla** [comment]
ok. @lolodomo: his LGTM and for me can be merged

**Pull request merged**